src/main/java/org/zalando/problem/spring/web/advice/validation/ConstraintViolationProblem.java

```diff
@@ -13,30 +10,28 @@
 import java.util.List;
 
 @Immutable
-@JsonTypeName(ConstraintViolationProblem.TYPE_VALUE)
 public final class ConstraintViolationProblem extends ThrowableProblem {
 
     public static final String TYPE_VALUE = "https://zalando.github.io/problem/constraint-violation";
     public static final URI TYPE = URI.create(TYPE_VALUE);
 
+    private final URI type;
     private final StatusType status;
-    private final String detail;
     private final List<Violation> violations;
 
     public ConstraintViolationProblem(final StatusType status, final List<Violation> violations) {
-        this(status, null, new ArrayList<>(violations));
+        this(TYPE, status, new ArrayList<>(violations));
     }
 
-    @JsonCreator
-    ConstraintViolationProblem(final StatusType status, @Nullable final String detail, final List<Violation> violations) {
+    ConstraintViolationProblem(final URI type, final StatusType status, final List<Violation> violations) {
```

Comment:
Deserializing into ConstraintViolationProblem was neither fully implemented nor documented. Also don't think it's a good idea. As stated in the definition of the type (see type URI) it's not meant to be used in production and rather for developer convenience, so deserialization should not be needed, the default implementation for problem should be sufficient.